doc/grids.rst

```diff
@@ -304,6 +305,103 @@ Xgcm can also deal with complex topologies such as cubed-sphere and
 lat-lon-cap.
 This is described in the :ref:`grid_topology` page.
 
+
+.. _Boundary conditions:
+
+Boundary conditions
+~~~~~~~~~~~~~~~~~~~
+
+For variables located at the boundaries, some operations need boundary conditions.
+Let's use the previous example axis, with center and left points::
+
+
+    |     |     |     |     |
+    U--T--U--T--U--T--U--T--|
+    |     |     |     |     |
+
+
+If we have a variable at the U (left) points, we have a problem for some operation
+(e.g. differentiating): how to treat the last T point?
+The solution is to add an extra point for the computation ('X' point on the following sketch)::
+
+
+    |     |     |     |     |
+    U--T--U--T--U--T--U--T--X
+    |     |     |     |     |
+
+
+Different options are possible (``fill`` this extra values with a certain number,
+``extend`` to the nearest value, ``extrapolate`` linearly using the 2 nearest points,
+or periodic condition if the grid axis is periodic).
+Attention, this boundary condition is used to give the value of X, not to give the value of the 
+boundary T point after the operation.
+
+We can illustrate it by creating some data located at the U point:
+
+.. ipython:: python
+
+    g = xr.ufuncs.sqrt(ds.x_g + 0.5) + xr.ufuncs.sin((ds.x_g - 0.5) * 2 * np.pi / 8)
+    g
+
+We show here the value of the extra added point for 5 cases (extended, extrapolated, filled with 0, filled with 5,
+and periodic). The periodic condition is not an argument of the methods, but is provided
+as an argument of the ``xgcm.Grid``. We will thus also create 2 grids: one periodic and another one not periodic.
+
+.. ipython:: python
+
+    def plot_bc(ds):
+        plt.plot(ds.x_g, g, marker='o', color='C6', label='g')
```

Comment:
This code block (and the following), do not show up nicely in the Readthedocs preview (https://xgcm--325.org.readthedocs.build/en/325/grids.html) because the lines are too long. 

See this example:
<img width="836" alt="image" src="https://user-images.githubusercontent.com/14314623/117825588-6f995100-b23d-11eb-9c3f-8729a08a7dc4.png">

Could you shorten the lines? Ideally you would use [black](https://github.com/psf/black) to format each code cell and then paste it back? I also wonder if there is a way to do this automatically. Ill open an issue for that in the future.

Reply:
I just tried to use black (and copy/paste), but it does not seem to solve the problem
![xgcm-doc-black](https://user-images.githubusercontent.com/18579092/117828621-447d2480-b272-11eb-86fd-d318b7051af2.png)